### ISSUE TITLE
Bug with `gatsby-plugin-mdx` and multiple `gatsby-remark-*` plugins

### ISSUE THREAD

**Issue by vincentdnl** [gatsbyjs/gatsby#21727]
## Description

When I do the following steps an error occur during the building of the bundle. 

### Steps to reproduce

1. `gatsby new reproducing-bug https://github.com/gatsbyjs/gatsby-starter-blog-theme` creating a new site from the blog theme starter.
2. `cd reproducing-bug`
3. Add a mdx page like `src/pages/about.mdx` and add some content in it.
4. `gatsby-remark-external-links` or `yarn add gatsby-remark-color-highlight`. I've tried with these two plugins and had the same error. I don't know if the problem comes from them or from `gatsby-plugin-mdx`
5. Add this config under `plugins` in `gatsby-config.js` (this comes from `gatsby-theme-blog-core`): 

```javascript
{
    resolve: `gatsby-plugin-mdx`,
    options: {
        extensions: [`.mdx`, `.md`],
        gatsbyRemarkPlugins: [
            {
                resolve: `gatsby-remark-images`,
                options: {
                    // should this be configurable by the end-user?
                    maxWidth: 1380,
                    linkImagesToOriginal: false,
                },
            },
            {resolve: `gatsby-remark-copy-linked-files`},
            {resolve: `gatsby-remark-smartypants`},
            {resolve: `gatsby-remark-external-links`}, // Adding this gatsby-remark plugin
        ],
        remarkPlugins: [require(`remark-slug`)],
    },
}
```

6. `yarn develop` should trigger the error

### Expected result

The bundle should build properly with no error.

### Actual result

I'm getting an error:

```
 ERROR #98123  WEBPACK

Generating development JavaScript bundle failed

unknown: Identifier '_frontmatter' has already been declared (100:13)

   98 | MDXContent.isMDXComponent = true;
   99 | 
> 100 | export const _frontmatter = {};
      |              ^
  101 | 

File: src/pages/about.mdx:100:13

failed Building development bundle - 2.828s
```

### Environment

```
  System:
    OS: Linux 5.3 Ubuntu 19.10 (Eoan Ermine)
    CPU: (4) x64 Intel(R) Core(TM) i5-6600 CPU @ 3.30GHz
    Shell: 5.7.1 - /usr/bin/zsh
  Binaries:
    Node: 12.13.0 - ~/.nvm/versions/node/v12.13.0/bin/node
    Yarn: 1.10.1 - /usr/bin/yarn
    npm: 6.12.0 - ~/.nvm/versions/node/v12.13.0/bin/npm
  Languages:
    Python: 2.7.17 - /usr/bin/python
  Browsers:
    Chrome: 62.0.3202.94
    Firefox: 73.0
  npmPackages:
    gatsby: ^2.19.7 => 2.19.7 
    gatsby-remark-color-highlight: ^1.0.2 => 1.0.2 
    gatsby-theme-blog: ^1.2.35 => 1.2.35 
```

**If you feel like it belongs somewhere else, I would gladly close the issue and open it in another repo!**

**Comment by LekoArts** [comment]
Hi!

Please check if this is the same bug: https://github.com/gatsbyjs/gatsby/issues/15486#issuecomment-510153237

So add them to `plugins` and see if it works. Thanks!

**Comment by vincentdnl** [comment]
Hi LekoArts, thanks for your reply! I tried the solution in the issue you mentioned and it doesn't work, I still get the same error.

**Comment by LekoArts** [comment]
Wait, where do you add this? Do you use `mdxOtherwiseConfigured` as `true` and have your own config for `gatsby-plugin-mdx` in your gatsby-config.js?

Please show a reproduction repository, would help a lot. Thanks!

**Comment by vincentdnl** [comment]
I've tried using `mdxOtherwiseConfigured` as `true` but it didn't change the error. I configured `gatsby-plugin-mdx` in my `gatsby-config.js` file.

Repository reproducing the bug: https://github.com/vincentdnl/bug-gatsby-mdx-remark-plugins

Note that it only affect pages. If I remove `src/pages/about.mdx` the bug disappears.

**Comment by mxmzb** [comment]
Looks, like the current fix is https://github.com/ChristopherBiscardi/gatsby-mdx/issues/354#issuecomment-529953977

`gatsby-plugin-mdx` just doesn't play well with themes at the moment.

**Comment by wartburggraaf** [comment]
> Looks, like the current fix is [ChristopherBiscardi/gatsby-mdx#354 (comment)](https://github.com/ChristopherBiscardi/gatsby-mdx/issues/354#issuecomment-529953977)
> 
> `gatsby-plugin-mdx` just doesn't play well with themes at the moment.

Unfortunately, I couldn't solve it with the above suggestion. I got the same error `unknown: Identifier '_frontmatter' has already been declared (1646:13)` while I was adding this into my project `https://theme-ui.com/guides/linked-headings/`
